### ISSUE TITLE
"Google Sitemaps" module: not listing cms pages & category paginated pages

### ISSUE THREAD

**Issue by guillaumeDIGITAL03** [PrestaShop/PrestaShop#33845]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Module v. 4.2.0 does not list:
- categories with paginated url (i.e. https://www.domain.com/123-category?page=2)
- CMS pages, neither categories (i.e https://www.domain.com/content/category/111-category-slug) nor pages (i.e https://www.domain.com/content/7-page-slug)

### Expected behavior

The module should naturally list all relevant urls generated by Prestashop.

### Steps to reproduce

1. Go to any website with cms pages and paginated category pages
2. Copy these urls
3. Go to /robots.txt page of the domain and get the sitemap url
4. Search each of the test urls listed on "1." through a paste within "Find" function of your browser

### PrestaShop version(s) where the bug happened

1.7.8.7

### PHP version(s) where the bug happened

7.3.33-4+0~20220627.98+debian9~1.gbp40b3e4

### If your bug is related to a module, specify its name and its version

Google Sitemaps v4.2.0

**Comment by sallemiines** [comment]
Hello @guillaumeDIGITAL03 

Could you please check this issue #30221 seems similar to yours.
Waiting for your feedback.

Thanks!

**Comment by Hlavtox** [comment]
@sallemiines Ines, put Waiting for PM please. This needs to be validated as it's quite important topic. Quick google gave me a mixed feedback. 👍

**Comment by LarsGMortensen** [comment]
Sorry for sticking my nose into this, but: Do you even want the XML-sitemap to include the paginated pages (page no. > 1) of categories? Doesn't those just create SEO-issues with "duplicate content" ?

**Comment by Hlavtox** [comment]
@LarsGMortensen That would not be a problem, paginated pages contain unique products and have canonicals to themselves. We consider them as unique pages to be indexed, if search engines decide to do so. However, I don't think I ever saw a paginated page coming up in the search results, google probably merges it to the main one anyway.

Checking the discussions online, like here - https://www.reddit.com/r/TechSEO/comments/10gfiob/should_pagination_pages_be_included_in_your/, **I don't think we should include paginated urls in the sitemap.**

Reasoning:
- Although these pages will get crawled and acknowledged by search engines, we don't want to spend the crawl budget on these pages, as they are not really important.
- Given the dynamic nature of ecommerce, the content of paginated pages could not match the SERP result anyway.
- Then there is the performance and implementation problem. If you want to include paginated pages, you would need to calculate the product count for every category and every special listing. And there could be infinite amount of search providers responsible for listing, so you would need to initiate product search on each of these categories and pages.

**Comment by kpodemski** [comment]
What's the point? Those are not unique pages, and products are naturally listed as separate entities in the sitemap

**Comment by LarsGMortensen** [comment]
@Hlavtox I fully agree: I don't think paginated urls should be in the sitemap either.

Also, I just did some tests, and there is a SEO-issue: If you fill in a text in the 'additional description' for a category, this text will also be included on the paginated pages. And as the paginated pages has canonical = themselves, it becomes duplicate content.
Note that the 'description' is only shown on the first page, not the paginated pages, while 'additional description' is also included on the paginated plages. 
Solution: 'description' and 'additional description' should only be shown on the first page, not the paginated pages.

**Comment by kpodemski** [comment]
@guillaumeDIGITAL03 do you have any research/content/resources which could prove that they should be included?

**Comment by kpodemski** [comment]
@LarsGMortensen 

Secondary description, in Classic, is only displayed on the first page:

https://github.com/PrestaShop/classic-theme/blob/e492757a0c184da864980659ecde66d484c39148/templates/catalog/_partials/category-footer.tpl#L26

**Comment by LarsGMortensen** [comment]
hi @kpodemski 
Yes, I also found that line of code in the template. However, its not working as expected:

![chrome_dP5CZcWzmK](https://github.com/PrestaShop/PrestaShop/assets/26773745/ff5f426c-98dc-45cb-a66e-91dfdd3725b5)

**Comment by LarsGMortensen** [comment]
@kpodemski I created a ticket about this bug:
https://github.com/PrestaShop/PrestaShop/issues/33903

**Comment by guillaumeDIGITAL03** [comment]
> Sorry for sticking my nose into this, but: Do you even want the XML-sitemap to include the paginated pages (page no. > 1) of categories? Doesn't those just create SEO-issues with "duplicate content" ?

Since I retrieve personnalized TITLE tand H1, it is not an SEO problem anymore.

**Comment by guillaumeDIGITAL03** [comment]
> 

- The module does not delete any xml sitemap.
- Launching the cron manually does not retrieve any error message. I do get the sitemap.
